packages/lib-runtime/src/store/PluginStore.ts

```diff
@@ -183,53 +110,76 @@ export class PluginStore<TLoadedExtension extends LoadedExtension = LoadedExtens
     };
   }
 
-  private invokeListeners(eventTypes: PluginStoreEventType[]) {
+  private invokeListeners(eventTypes: PluginEventType[]) {
     eventTypes.forEach((t) => {
       this.listeners.get(t)?.forEach((listener) => {
         listener();
       });
     });
   }
 
-  /**
-   * Add new plugin to the `PluginStore`.
-   *
-   * Once added, the plugin is disabled by default. Enable it to put its extensions into use.
-   *
-   * Returns `true` if the plugin was added successfully.
-   */
-  addPlugin(metadata: PluginMetadata, processedExtensions: TLoadedExtension[]) {
-    const pluginName = metadata.name;
-    const pluginVersion = metadata.version;
+  getExtensions() {
+    return [...this.extensions];
+  }
 
-    if (this.loadedPlugins.has(pluginName)) {
-      consoleLogger.warn(`Attempt to re-add an already loaded plugin ${pluginName}`);
-      return false;
-    }
+  getPluginInfo() {
+    const loadedEntries = Array.from(this.loadedPlugins.keys()).reduce((acc, pluginName) => {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const plugin = this.loadedPlugins.get(pluginName)!;
```

Comment:
Could you do this instead to remove the need to avoid the `no-non-null-assertion`
```suggestion
    const loadedEntries = Array.from(this.loadedPlugins.values()).reduce((acc, plugin) => {
      const { pluginName } = plugin.metadata;
```

Reply:
ESLint rules can't be 100% accurate and it's perfectly fine to suppress them if necessary.

That said - agreed, will do the change :smiley: